third_party/terraform/resources/resource_google_billing_subaccount.go

```diff
@@ -0,0 +1,134 @@
+package google
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/cloudbilling/v1"
+)
+
+func resourceBillingSubaccount() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceBillingSubaccountCreate,
+		Read:   resourceBillingSubaccountRead,
+		Delete: resourceBillingSubaccountDelete,
+		Update: resourceBillingSubaccountUpdate,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"display_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"open": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"master_billing_account": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if new == canonicalBillingAccountName(old) {
+						return false
+					}
+					return true
+				},
+			},
+			"billing_account_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"rename_on_destroy": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+func resourceBillingSubaccountCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	billingAccount := new(cloudbilling.BillingAccount)
+	displayName := d.Get("display_name").(string)
+	masterBillingAccount := d.Get("master_billing_account").(string)
+	billingAccount.DisplayName = displayName
+	billingAccount.MasterBillingAccount = canonicalBillingAccountName(masterBillingAccount)
+
+	resp, err := config.clientBilling.BillingAccounts.Create(billingAccount).Do()
+	if err != nil {
+		return fmt.Errorf("Error creating billing subaccount '%s' in master account '%s': %s", displayName, masterBillingAccount, err)
+	}
+
+	d.SetId(resp.Name)
+	d.Set("name", resp.Name)
+	d.Set("open", resp.Open)
+	d.Set("billing_account_id", strings.TrimPrefix(d.Get("name").(string), "billingAccounts/"))
+
+	return nil
+}
+
+func resourceBillingSubaccountRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	id := d.Id()
+
+	var billingAccount *cloudbilling.BillingAccount
+	resp, err := config.clientBilling.BillingAccounts.Get(d.Id()).Do()
```

Comment:
```suggestion
	res, err := config.clientBilling.BillingAccounts.Get(d.Id()).Do()
```

Reply:
Bit nitpicky, but res is used more commonly in the provider. You probably don't want to apply this directly, since the references need to be changed too.